### PR TITLE
Implement LSF_RESOURCE and EXCLUDE_HOST queue option

### LINKED ISSUE #7345: Implement LSF_RESOURCE queue option for the LSF_driver
See the old JobQueue usage.

### PR DESCRIPTION
**Pull request by jonathan-eq**
**Issue**
Resolves #7345 

**Approach**
This PR makes scheduler lsf_driver use QUEUE_OPTION LSF_RESOURCE as qsub resource_request option.
It also adds the EXCLUDE_HOST queue option for LSF, as they were closely related in the legacy C driver.

(Screenshot of new behavior in GUI if applicable)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/scheduler/__init__.py [hunk 1]
```diff
@@ -38,6 +38,8 @@ def create_driver(config: QueueConfig) -> Driver:
             bsub_cmd=queue_config.get("BSUB_CMD"),
             bkill_cmd=queue_config.get("BJOBS_CMD"),
             bjobs_cmd=queue_config.get("BJOBS_CMD"),
+            queue_name=queue_config.get("LSF_QUEUE"),
```
Comment (berland) [inline src/ert/scheduler/__init__.py:41]: Is it a bug that we did not pass in LSF_QUEUE already? (if so, it should probably be a separate PR, with some testing)

Comment (jonathan-eq) [reply src/ert/scheduler/__init__.py:41]: It seems like it. I will split this commit out into a new PR, and create an issue for it.

**Comment by berland** [comment]
From `man bsub`:

```
       -R "res_req" [-R "res_req" ...]
                Runs the job on a host that meets the specified resource
                requirements. A resource requirement string describes
                the resources a job needs. LSF uses resource
                requirements to select hosts for job execution. Resource
                requirement strings can be simple (applying to the
                entire job), compound (applying to the specified number
                of slots), or alternative.

```
which means we should probably name the variable in the driver the same, "`resource_requirement`"

File: src/ert/scheduler/lsf_driver.py [hunk 2]
```diff
@@ -150,10 +152,14 @@ async def submit(
             script_path = Path(script_handle.name)
         assert script_path is not None
         script_path.chmod(script_path.stat().st_mode | stat.S_IEXEC)
+        arg_resource_request = (
+            ["-R", f"'{self._resource_request}'"] if self._resource_request else []
```
Comment (berland) [inline src/ert/scheduler/lsf_driver.py:156]: Is the single quoting here needed? Should it be responsibility of the user? Can the quotes even be injected via the config?

File: tests/unit_tests/scheduler/test_lsf_driver.py [hunk 3]
```diff
@@ -130,6 +130,15 @@ async def test_submit_with_default_queue():
     assert "-q" not in Path("captured_bsub_args").read_text(encoding="utf-8")
 
 
+@pytest.mark.usefixtures("capturing_bsub")
+async def test_submit_with_resource_request():
```
Comment (berland) [inline tests/unit_tests/scheduler/test_lsf_driver.py:134]: I think we can afford an integration test toward the real cluster also on this. Just use some low requirements that seems relevant and write test function similiar to test_submit in test_generic_driver that writes the string test to a file, and then assert that the string can be found.

**Comment by jonathan-eq** [comment]
> From `man bsub`:
> 
> ```
>        -R "res_req" [-R "res_req" ...]
>                 Runs the job on a host that meets the specified resource
>                 requirements. A resource requirement string describes
>                 the resources a job needs. LSF uses resource
>                 requirements to select hosts for job execution. Resource
>                 requirement strings can be simple (applying to the
>                 entire job), compound (applying to the specified number
>                 of slots), or alternative.
> ```
> 
> which means we should probably name the variable in the driver the same, "`resource_requirement`"

It was named resource_request in the old driver, but I think it is a good improvement.

File: src/ert/scheduler/lsf_driver.py (hunk 2 above)
Comment (jonathan-eq) [reply src/ert/scheduler/lsf_driver.py]: It seems to do so in the old driver. Is LSF_SUBMIT_REMOTE_SHELL the submit method we will be using, or the one we are dropping with the scheduler?
```
char *quoted_resource_request = NULL;
    if (req) {
        if (driver->submit_method == LSF_SUBMIT_REMOTE_SHELL) {
            quoted_resource_request = saprintf("\"%s\"", req);
            free(req);
        } else
            quoted_resource_request = req;
    }
    return quoted_resource_request;
```

Comment (berland) [reply src/ert/scheduler/lsf_driver.py]: We are dropping the remote shell feature yes.

If there can be spaces in resource requests, it should be tested in integration tests.

File: src/ert/scheduler/__init__.py [hunk 4]
```diff
@@ -38,7 +38,9 @@ def create_driver(config: QueueConfig) -> Driver:
             bsub_cmd=queue_config.get("BSUB_CMD"),
             bkill_cmd=queue_config.get("BKILL_CMD"),
             bjobs_cmd=queue_config.get("BJOBS_CMD"),
+            exclude_hosts=queue_config.get("EXCLUDE_HOST", "").split(","),
```
Comment (xjules) [inline src/ert/scheduler/__init__.py:41]: Does it work?

Comment (jonathan-eq) [reply src/ert/scheduler/__init__.py:41]: Yes

**Review by xjules — APPROVED** [review]
Looks very nice! Let's ship it :rocket:
